source/manual/architecture-deep-dive.html.md

```diff
@@ -0,0 +1,462 @@
+title. Sometimes a single change can trigger changes in thousands of items.
+
+Putting both the directly changed and indirectly changed content items
+on the same queue would mean it would take a long time to see the changes
+in a document you've edited. Generally, it is less important to see a quick
+change to the indirectly changed content than it is to see a change in the
+directly changed content items. Therefore we have a concept of 'high' and
+'low' priority queues.
+
+The main content item is processed in the high priority queue. Exactly the
+same things happen to the low priority content items as the high priority
+content items; it just tends to take longer as there are more items to
+process.
+
+The process for finding the content items affected by a content change is
+known as [dependency resolution]. Content items can be associated with
+other content items in a number of ways. For example, you may provide an
+[array of organisation IDs][schema-organisations-example] in your payload
+when sending to publishing API, to indicate that those organisations are
+responsible for the content (this is stored on the content item in content
+store as `links.organisations`).
+
+Content can also be tagged to [taxonomies], which are used to describe where
+in the site hierarchy the content sits. These are stored on the content item
+as `links.taxons`. Some apps have their own interface for tagging, or you can
+tag content independently using [content-tagger].
+
+[content-tagger]: https://github.com/alphagov/content-tagger
+[dependency resolution]: https://github.com/alphagov/publishing-api/blob/master/doc/dependency-resolution.md
+[schema-organisations-example]: https://github.com/alphagov/govuk-content-schemas/blob/d9684140462e4a138668539c04829cd808636ed5/dist/formats/news_article/publisher_v2/schema.json#L70-L73
+[taxonomies]: /manual/taxonomy.html
+
+### Summary
+
+The publishing app uses the publishing API to create and synchronise a new
+edition of a document, which consolidates related content items into it prior to
+sending to content store. All affected content items are added to a publishing
+queue, which triggers downstream actions such as cache clearing and email alerts.
+
+## What happens when a developer deploys a change to an application?
+
+### Environments
+
+Everything you've read about the live and draft stacks, you can now multiply
+threefold, as they each exist in the following environments:
+
+- Production
+- Staging
+- Integration
+
+Data is copied from Production to Staging - and from Staging to Integration - every
+24 hours via [automated Jenkins jobs][copy-data-to-staging]. This way our
+environments are always roughly in sync, although it's worth noting that email
+addresses are anonymised and access-limited documents are obfuscated before data is
+copied. The data copying is partly configured in [env-sync-and-backup], however
+we're gradually [configuring this in govuk-puppet][govuk-env-sync] in the move to AWS.
```

Comment:
Given most data syncing now happens through govuk-env-sync, it might be worth flipping the relevance in this paragraph. I realise this was probably correct when you wrote it a while ago.